src/pytorch_lightning/strategies/parallel.py

```diff
@@ -89,10 +89,21 @@ def all_gather(self, tensor: Tensor, group: Optional[Any] = None, sync_grads: bo
         """Perform a all_gather on all processes."""
         return all_gather_ddp_if_available(tensor, group=group, sync_grads=sync_grads)
 
-    def reduce_boolean_decision(self, decision: bool) -> bool:
+    def reduce_boolean_decision(self, decision: bool, all: bool = True) -> bool:
+        """Reduces a boolean decision over distributed processes. By default is analagous to ``all`` from the
+        standard library, returning ``True`` only if all input decisions evaluate to ``True``. If ``all`` is set to
+        ``False``, it behaves like ``any`` instead.
```

Comment:
I am wondering, are there any other logical operations we need to support in the future? If yes, we should make this a parameter that takes the actual op, i.e., `op: Literal["all", "any"]`. Probably not, but just for consideration.

Reply:
Interesting idea! I can't think of other ops that directly map and would make sense in this case so probably makes sense to keep it a bool here. I agree that's a good approach to keep in mind in the future and I'm open to switching things here others feel strongly about it.